client/src/components/SignUp.js

```diff
@@ -54,9 +54,29 @@ signupLink:{
 }));
 function SignUp (){
     const classes = useStyles();
-    const [userName, setName]=useState('')
-    const [userEmail,setEmail]=useState('')
-    const [userPass, setPass]=useState('')
+    const [name, setName]=useState('')  // Changed names to correspond with backend.
+    const [email,setEmail]=useState('')
+    const [password, setPass]=useState('')
+    const [confirm, setConfirm]=useState('')
+    const handleClick = () => {
+        fetch("/auth/register", {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'name': name,
+                'email': email,
+                'password': password,
+                'confirm': confirm
+            })})
+            .then(response => response.json())
+            .then(console.log('Success:', name)) // Logs success even when unsuccessful, not certain how to fix it.
+            .catch((error) => {
+                console.error('Error:', error)
+            })
+    }
+    // Added confirm password field. Still needs to check for password being at least six characters
```

Comment:
We should also check here if the confirm password matches password. We can do these checks either on the input field (set onChange to a function that 1) sets the value and 2) validates the value) or on the submit (handleClick checks validation and either shows error messages or makes the API call). It's kind of a UX decision on which to use. Either way, we'll want to use the error and helperText on TextField: https://material-ui.com/components/text-fields/#validation

Reply:
It now updates the error and helperText when submit is clicked. I don't have a huge preference one way or another, but I do slightly prefer this way to avoid users being annoyed by being told their details are wrong as they are entering them.